### ISSUE TITLE
Loss spike when training mosaic-bert (fp32)

### ISSUE THREAD

**Issue by sameerreddy13** [mosaicml/examples#236]
I am training with the `mosaic-bert-base-uncased.yaml` recipe on 8xA40s, with data created with the mosaic provided script for c4. I consistently get a loss spike and loss being stuck around 10k -15k steps into training. The only change is using fp32 instead of bfloat16.

Below spikes at ~10k
<img width="1324" alt="Screenshot 2023-03-14 at 9 14 53 AM" src="https://user-images.githubusercontent.com/14104824/225012214-90a3dd6d-a491-4853-ba6c-70fb4b89ffb3.png">
<img width="1339" alt="Screenshot 2023-03-14 at 9 15 13 AM" src="https://user-images.githubusercontent.com/14104824/225012296-daf32196-7be7-4690-af16-e832679e59e1.png">

**Comment by sameerreddy13** [comment]
For my environment you can refer to my other issue (https://github.com/mosaicml/examples/issues/237#issuecomment-1468506231)

**Comment by jacobfulano** [comment]
Hey @sameerreddy13, 

We did most of our benchmarking and testing in bf16, and chose hyperparameters that worked well with this setting.

With regards to fp32, we would recommend lowering the learning rate slightly as a first line of defense (barring potential issues relating to environment). If you are also playing around with a larger architecture (e.g. BERT-Large), we would also recommend lowering the learning rate!

**Comment by sameerreddy13** [comment]
Hi @jacobfulano. Thank you for the advice! I actually realized the issue was with my flash attention being broken and am resolving that in another issue.
